test/e2e/framework/helpers/yunikorn/rest_api_utils.go

```diff
@@ -106,6 +106,51 @@ func (c *RClient) GetHealthCheck() (dao.SchedulerHealthDAOInfo, error) {
 	return healthCheck, err
 }
 
+func (c *RClient) WaitforQueueToAppear(partition string, queueName string, timeout int) error {
+	var queueInfo = wait.PollImmediate(300*time.Millisecond, time.Duration(timeout)*time.Second, c.IsQueuePresent(partition, queueName))
+	startTime := time.Now().Unix()
+	endTime := startTime + int64(timeout)
+	// if the queue is not present , Polling for timeout seconds
+	for {
+		currentTime := time.Now().Unix()
+		if queueInfo != nil {
+			if currentTime <= endTime {
+				queueInfo = wait.PollImmediate(300*time.Millisecond, time.Duration(timeout)*time.Second, c.IsQueuePresent(partition, queueName))
+			} else {
+				break
+			}
+		} else {
+			break
+		}
+	}
```

Comment:
To me, this part is very unclear. We already do multiple calls inside `PollImmediate()`, why do we do it again in a for loop? It's very confusing, especially the nesting.

Also, "queueInfo" is an error object, can't we just call it "err"?